model/src/main/java/org/projectnessie/model/Namespace.java

```diff
@@ -92,8 +91,9 @@ public static Namespace of(String... elements) {
           String.format(ERROR_MSG_TEMPLATE, Arrays.toString(elements)));
     }
 
-    String name = String.join(DOT, Arrays.asList(elements));
-    return ImmutableNamespace.builder().name(name).id(name).build();
+    List<String> list = Arrays.asList(elements);
+    String name = String.join(DOT, list);
+    return ImmutableNamespace.builder().name(name).id(name).elements(list).build();
```

Comment:
we could most likely, I just didn't see it in the scope of this PR itself

Reply:
From my POV is belongs in this PR, because we're standardizing how we parse names and add `elements`... might as well localize all of that logic in this class to avoid setting `elements` and `name` separately to unrelated values (by mistake, of course).

Reply:
done